src/mgmt/tests/srv_drpc_tests.c

```diff
@@ -220,9 +220,58 @@ test_drpc_pool_get_acl_success(void **state)
 	}
 }
 
+/*
+ * dRPC List Pools setup/teardown
+ */
+
+static int
+drpc_list_pools_setup(void **state)
+{
+	/* mock_ds_mgmt_list_pools_setup(); */
+
+	return 0;
+}
+
+static int
+drpc_list_pools_teardown(void **state)
+{
+	/* mock_ds_mgmt_list_pools_teardown(); */
+
+	return 0;
+}
+
+/*
+ * dRPC List Pools tests
+ */
+static void
+test_drpc_list_pools_bad_request(void **state)
+{
+	Drpc__Call	call = DRPC__CALL__INIT;
+	Drpc__Response	resp = DRPC__RESPONSE__INIT;
+	uint8_t		bad_bytes[16];
+	size_t		i;
+
+	/* Fill out with junk that won't translate to a ListPoolsReq */
+	for (i = 0; i < sizeof(bad_bytes); i++)
+		bad_bytes[i] = i;
+
+	call.body.data = bad_bytes;
+	call.body.len = sizeof(bad_bytes);
+
+	ds_mgmt_drpc_list_pools(&call, &resp);
+
+	assert_int_equal(resp.status, DRPC__STATUS__FAILURE);
+	assert_null(resp.body.data);
+	assert_int_equal(resp.body.len, 0);
+}
```

Comment:
IMO it would be valuable and low lift to add a few more tests similar to those for get ACL.

Reply:
Yes, agreed and this is the plan to add more tests in a subsequent commit.